nipype/algorithms/stats.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+'''
+Algorithms to compute statistics on :abbr:`fMRI (functional MRI)`
+
+    Change directory to provide relative paths for doctests
+    >>> import os
+    >>> filepath = os.path.dirname(os.path.realpath(__file__))
+    >>> datadir = os.path.realpath(os.path.join(filepath, '../testing/data'))
+    >>> os.chdir(datadir)
+
+'''
+from __future__ import (print_function, division, unicode_literals,
+                        absolute_import)
+from builtins import str
+
+import numpy as np
+import nilearn.input_data as nl
+
+from .. import logging
+from ..interfaces.base import (traits, TraitedSpec, BaseInterface,
+                               BaseInterfaceInputSpec, File)
+IFLOG = logging.getLogger('interface')
+
+class SignalExtractionInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc='4-D fMRI nii file')
+    label_file = File(exists=True, mandatory=True,
+                      desc='a 3-D label image, with 0 denoting background, or '
+                      'a 4-D file of probability maps.')
+    class_labels = traits.List(mandatory=True,
```

Comment:
Another example (which is not an unlikely use case) is the FreeSurfer parcellation -> https://surfer.nmr.mgh.harvard.edu/fswiki/FsTutorial/AnatomicalROI/FreeSurferColorLUT

I think we should allow the user reuse the labels from the segmentation --what if the labels are not consecutive numbers?--